.pipelines/prchecks/CveSpecFilePRCheck/CveSpecFilePRCheck.py

```diff
@@ -284,70 +284,153 @@ def get_package_directory_files(spec_path: str) -> List[str]:
+        file_list = get_package_directory_files(spec_file)
+        package_name = extract_package_name(spec_content, spec_file)
         
-        # Early exit if no spec files changed
-        if not changed_spec_files:
-            return [], ai_analysis, False
-            
-        # Process each changed spec file for anti-patterns
-        for spec_path in changed_spec_files:
-            logger.info(f"Running anti-pattern detection on: {spec_path}")
-            
-            spec_content = get_spec_file_content(spec_path)
-            if not spec_content:
-                logger.warning(f"Could not read spec file content for {spec_path}, skipping detailed analysis")
-                continue
-                
-            file_list = get_package_directory_files(spec_path)
-            
-            # Detect anti-patterns
-            anti_patterns = detector.detect_all(spec_path, spec_content, file_list)
-            
-            all_anti_patterns.extend(anti_patterns)
-            
-            # Log detected issues
-            if anti_patterns:
-                critical_count = sum(1 for p in anti_patterns if p.severity >= Severity.ERROR)
-                warning_count = sum(1 for p in anti_patterns if p.severity == Severity.WARNING)
-                
-                logger.warning(f"Found {len(anti_patterns)} anti-patterns in {spec_path}:")
-                logger.warning(f"   - {critical_count} critical/error issues")
-                logger.warning(f"   - {warning_count} warnings")
-            else:
-                logger.info(f"No anti-patterns detected in {spec_path}")
+        # Create result container for this spec
+        spec_result = SpecFileResult(
+            spec_path=spec_file,
+            package_name=package_name
+        )
         
-        return all_anti_patterns, ai_analysis, False
+        # Run anti-pattern detection
+        analyzer = AntiPatternDetector()
+        spec_result.anti_patterns = analyzer.detect_all(
+            spec_file, spec_content, file_list
+        )
         
-    except Exception as e:
-        logger.error(f"Error in analyze_spec_files: {str(e)}", exc_info=True)
-        return all_anti_patterns, ai_analysis, True
+        # Run AI analysis if enabled and configured
+        if os.environ.get("ENABLE_AI_ANALYSIS", "false").lower() == "true":
+            try:
+                openai_client = _initialize_openai_client()
+                if openai_client:
+                    # Get AI analysis for this specific spec
+                    spec_ai_analysis = call_openai_for_single_spec(
+                        openai_client, spec_file, spec_content, diff_text
+                    )
+                    spec_result.ai_analysis = spec_ai_analysis
+            except Exception as e:
+                logger.warning(f"AI analysis failed for {spec_file}: {e}")
+        
+        result.spec_results.append(spec_result)
+    
+    # Trigger post-init calculations
+    result.__post_init__()
```

Comment:
Could you do
```
return MultiSpecAnalysisResult(
    spec_results = spec_result_list # gather this in the loop
    )   
```

I *think* this should auto run the calculations with all the data in place?